src/orion/core/worker/primary_algo.py

```diff
@@ -137,48 +138,68 @@ def suggest(self, num: int) -> list[Trial] | None:
         New parameters must be compliant with the problem's domain `orion.algo.space.Space`.
 
         """
-        transformed_trials = self.algorithm.suggest(num)
-
-        if transformed_trials is None:
-            return None
 
         trials: list[Trial] = []
-        for transformed_trial in transformed_trials:
-            if transformed_trial not in self.transformed_space:
-                raise ValueError(
-                    f"Trial {transformed_trial.id} not contained in space:\n"
-                    f"Params: {transformed_trial.params}\n"
-                    f"Space: {self.transformed_space}"
-                )
-            original = self.transformed_space.reverse(transformed_trial)
-            if original in self.registry:
-                logger.debug(
-                    "Already have a trial that matches %s in the registry.", original
-                )
-                # We already have a trial that is equivalent to this one.
-                # Fetch the actual trial (with the status and possibly results)
-                original = self.registry.get_existing(original)
-                logger.debug("Matching trial (with results/status): %s", original)
 
-                # Copy over the status and results from the original to the transformed trial
-                # and observe it.
-                transformed_trial = _copy_status_and_results(
-                    original_trial=original, transformed_trial=transformed_trial
-                )
+        for suggest_attempt in range(1, self.max_suggest_attempts + 1):
```

Comment:
I think we should add a test for this new behavior. Not in the test-suite but in the wrapper's set of tests.

Reply:
Fixed in bb73b8272eef6e04aa98230f8680d94233a6ea3f